src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsImporter.kt

```diff
@@ -12,46 +12,38 @@ import java.time.Duration
 import java.time.LocalDateTime
 import java.util.concurrent.atomic.AtomicBoolean
 
-private const val COLUMN_HEADINGS = 1
-
 @Component
 class LocationsImporter(private val repo: LocationRepository,
                         private val clock: Clock,
                         private val schedule34LocationsProvider: Schedule34LocationsProvider) {
 
     private val logger = LoggerFactory.getLogger(javaClass)
-    private val rowOffset = 2
+
     private val running = AtomicBoolean(false)
 
     @Value("\${import-files.locations}")
     private lateinit var locationsFile: String
 
     fun import(locationsWorkbook: XSSFWorkbook) {
 
-        val errors: MutableList<LocationImportError?> = mutableListOf()
-
         var total = 0
 
-        LocationTab.values().map { locationTab ->
-            val sheet = locationTab.sheet(locationsWorkbook)
+        val spreadsheet = LocationsSpreadsheet(locationsWorkbook)
 
-            val rows = sheet.drop(COLUMN_HEADINGS).filterNot { it.getCell(1)?.stringCellValue.isNullOrBlank() }
-
-            rows.forEachIndexed{ index, row ->
-                run {
-                    Result.runCatching {
-                        locationTab.map(row).let {
-                            repo.save(it)
-                            total++
-                        }
-                    }.onFailure { errors.add(LocationImportError(locationTab, index + rowOffset, it.cause?.cause ?: it)) }
-                }
+        LocationsSpreadsheet.Tab.values().map { tab ->
+            spreadsheet.getRowsFrom(tab).forEach { row ->
+                Result.runCatching {
+                    spreadsheet.mapToLocation(row).let {
+                        repo.save(it)
+                        total++
```

Comment:
I couldn't see where you later reference `total`

Reply:
Line 46

Reply:
I think ultimately I'd like to change this...